pkg/cmd/repo/edit/edit.go

```diff
@@ -403,16 +408,26 @@ func interactiveRepoEdit(opts *EditOptions, r *api.Repository) error {
 				return err
 			}
 		case optionVisibility:
-			opts.Edits.Visibility = &r.Visibility
-			//nolint:staticcheck // SA1019: prompt.SurveyAskOne is deprecated: use Prompter
-			err = prompt.SurveyAskOne(&survey.Select{
-				Message: "Visibility",
-				Options: []string{"public", "private", "internal"},
-				Default: strings.ToLower(r.Visibility),
-			}, opts.Edits.Visibility)
+			opts.Edits.Visibility = new(string)
+			*opts.Edits.Visibility = r.Visibility
+
+			visibilityOptions := []string{"public", "private", "internal"}
+			selected, err := opts.Prompter.Select("Visibility", strings.ToLower(r.Visibility), visibilityOptions)
 			if err != nil {
 				return err
 			}
+			*opts.Edits.Visibility = visibilityOptions[selected]
```

Comment:
We could hold off with assignment to `opts.Edits.Visibility` until after the confirmation, so that we don't have to restore the original value if the confirmation was denied